torch/nn/parallel/distributed.py

```diff
@@ -380,14 +386,15 @@ def forward(self, *inputs, **kwargs):
         else:
             output = self.module(*inputs, **kwargs)
 
-        # We'll return the output object verbatim since it is a freeform object.
-        # We need to find any tensors in this object, though, because we need to
-        # figure out which parameters were used during this forward pass,
-        # to ensure we short circuit reduction for any unused parameters.
-        if self.find_unused_parameters:
-            self.reducer.prepare_for_backward(list(_find_tensors(output)))
-        else:
-            self.reducer.prepare_for_backward([])
+        if not self.accumulate_grads:
```

Comment:
Do you need to skip `_sync_params` as well? Otherwise, params in `self.modules_params[0]` will be broadcast to other replicas in every forward, which is not necessary if `self.accumulate_grads=True`, no?

Reply:
@mrshenli Right, I should skip that as well. And I should also document not calling optimizer.step when set to True